File: src/main/kotlin/pray/Madhab.java
package pray;

import pray.internal.ShadowLength;

/**
 * pray.Madhab for determining how Asr is calculated
 */
public enum Madhab {
    /**
     * Shafi pray.Madhab
     */
    SHAFI,

    /**
     * Hanafi pray.Madhab
     */
    HANAFI;

    ShadowLength getShadowLength() {
        switch (this) {
            case SHAFI -> {
                return ShadowLength.SINGLE;
            }
            case HANAFI -> {
                return ShadowLength.DOUBLE;
            }
            default -> {
                throw new IllegalArgumentException("Invalid pray.Madhab");
            }
        }
    }
}

File: src/main/kotlin/pray/Madhab.kt
package pray

import pray.internal.ShadowLength

/**
 * pray.Madhab for determining how Asr is calculated
 */
enum class Madhab {
    /**
     * Shafi pray.Madhab
     */
    SHAFI,

    /**
     * Hanafi pray.Madhab
     */
    HANAFI;

    val shadowLength: ShadowLength
        get() = when (this) {
            SHAFI -> {
                ShadowLength.SINGLE
            }

            HANAFI -> {
                ShadowLength.DOUBLE
            }

            else -> {
                throw IllegalArgumentException("Invalid pray.Madhab")
            }
        }
}
